addons/crm_mail_client_extension/controllers/main.py

```diff
@@ -3,34 +3,94 @@
+
+    @http.route('/mail_client_extension/lead/create_from_partner', type='http', auth='user', methods=['GET'])
+    def crm_lead_redirect_create_form_view(self, partner_id):
+        """
+            deprecated route, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        server_action = http.request.env.ref("crm_mail_client_extension.lead_creation_prefilled_action")
+        return werkzeug.utils.redirect('/web#action=%s&model=crm.lead&partner_id=%s' % (server_action.id, int(partner_id)))
+
+    def _get_leads(self, partner_id, limit=5, offset=0):
+        if not partner_id or partner_id <= 0:
+            return []
+
+        partner_leads = request.env['crm.lead'].search(
+            [('partner_id', '=', partner_id)], offset=offset, limit=limit)
+        recurring_revenues = request.env.user.has_group('crm.group_use_recurring_revenues')
+
         leads = []
         for lead in partner_leads:
-            leads.append({
+            lead_values = {
                 'id': lead.id,
                 'name': lead.name,
-                'expected_revenue': formatLang(request.env, lead.expected_revenue, monetary=True, currency_obj=lead.company_currency),
-            })
+                'expected_revenue': formatLang(request.env, lead.expected_revenue, monetary=True,
+                                               currency_obj=lead.company_currency),
+                'probability': lead.probability,
+            }
 
-        return {'leads': leads}
+            if recurring_revenues:
+                lead_values.update({
+                    'recurring_revenue': formatLang(request.env, lead.recurring_revenue, monetary=True,
+                                                    currency_obj=lead.company_currency),
+                    'recurring_plan': lead.recurring_plan.name,
+                })
 
-    @http.route('/mail_client_extension/lead/create_from_partner', type='http', auth='user', methods=['GET'])
-    def crm_lead_redirect_form_view(self, partner_id):
-        server_action = http.request.env.ref("crm_mail_client_extension.lead_creation_prefilled_action")
-        return werkzeug.utils.redirect('/web#action=%s&model=crm.lead&partner_id=%s' % (server_action.id, int(partner_id)))
+            leads.append(lead_values)
+
+        return leads
+
+    def _get_partner_extra_info(self, partner_id=None):
+        extra_info = super(MailClientExtensionController, self)._get_partner_extra_info(partner_id)
+        extra_info['leads'] = self._get_leads(partner_id)
+        return extra_info
+
+    def _get_loggable_models(self):
+        loggable_models = super(MailClientExtensionController, self)._get_loggable_models()
+        loggable_models.append('crm.lead')
+        return loggable_models
+
+    @http.route('/mail_client_extension/lead/create_from_email', type='json', auth='outlook', cors="*")
+    def create_lead_from_email(self, partner_id, email_body):
```

Comment:
To keep similar naming, I would go for "_crm_lead_create".
The fact that is "from an email" is not very necessary IMO.